From 783d3d380e6abf107e0747d229da0f5e001c623f Mon Sep 17 00:00:00 2001
From: Lars Kanis <lars@greiz-reinsdorf.de>
Date: Sun, 20 Dec 2020 19:08:22 +0100
Subject: [PATCH] Don't mask out build env vars for cross-ruby

This was introduced in commit a7c113e580c1565000166f9967da82ff2ecdaeaa to avoid interference with host build.
However it prohibits setting important flags for cross-build.

Also don't force 'LDFLAGS=-pipe -s' in configure, which doesn't work with clang targeting MacOS.
It can now be set externally.
---
 tasks/bin/cross-ruby.rake | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tasks/bin/cross-ruby.rake b/tasks/bin/cross-ruby.rake
index 8edf5cd..8317a2a 100644
--- a/tasks/bin/cross-ruby.rake
+++ b/tasks/bin/cross-ruby.rake
@@ -46,7 +46,7 @@ RUBY_SOURCE = ENV['SOURCE']
 RUBY_BUILD = RbConfig::CONFIG["host"]

 # Unset any possible variable that might affect compilation
-["CC", "CXX", "CPPFLAGS", "LDFLAGS", "RUBYOPT"].each do |var|
+["RUBYOPT"].each do |var|
   ENV.delete(var)
 end

--
2.27.0

